lib/net/ssh/transport/session.rb

```diff
@@ -61,12 +61,12 @@ def initialize(host, options={})
       @bind_address = options[:bind_address] || nil
       @options = options
 
-      debug { "establishing connection to #{@host}:#{@port}" }
-
       @socket =
         if (factory = options[:proxy])
+          debug { "establishing connection to proxy" }
```

Comment:
> It is completely ignored
Not sure what do you mean it's ignored, it's not ignored for sure. For example in the ssh command example you showed: `ssh -W host:port proxy.host.name` the command does connect to `host:port` through the intermediate host `porxy.host.name`, you might be connecting to `host:port` indirectly, but you're still connecting there (through a proxy).

The connection can fail if the proxy is not available or it can fail or the host/port you're connecting to through the proxy is not available.